### ISSUE TITLE
help wanted

### ISSUE THREAD

**Issue by hyuegithub** [intel/ros_object_analytics#48]
when I tried to launch the   it gets stuck at waiting for service  /ObjectAnalyticsMgr/load_nodelet,
waitForService: Service [/ObjectAnalyticsMgr/load_nodelet] has not been advertised, waiting...
 is there anything I missed while trying to use the package?anyone can help me?

**Comment by yechun1** [comment]
which OS do you use?  and would you please provide more error information?

**Comment by hyuegithub** [comment]
Thanks for your reply ,I am using Ubuntu16.04+ROS Kinetic+kinect1, After running the analytics_opencl_caffe.launch I always get the following error:

`I1127 18:44:19.747236 22046 net.cpp:280] Network initialization done.
I1127 18:44:19.747241 22046 net.cpp:281] Memory required for data: 86785584
[ INFO] [1543315459.928669138]: Load resources completely!
[object_analytics/ObjectAnalyticsMgr-1] process has died [pid 22046, exit code -8, cmd /opt/ros/kinetic/lib/nodelet/nodelet manager __name:=ObjectAnalyticsMgr __log:=/home/user/.ros/log/5ecdc050-f231-11e8-adc1-1c1b0d23f337/object_analytics-ObjectAnalyticsMgr-1.log].
log file: /home/user/.ros/log/5ecdc050-f231-11e8-adc1-1c1b0d23f337/object_analytics-ObjectAnalyticsMgr-1*.log
[ INFO] [1543315467.535394771]: Bond broken, exiting
[object_analytics/tracker-5] process has finished cleanly
log file: /home/user/.ros/log/5ecdc050-f231-11e8-adc1-1c1b0d23f337/object_analytics-tracker-5*.log
[ INFO] [1543315468.031738817]: Bond broken, exiting
[ INFO] [1543315468.031737320]: Bond broken, exiting
[ INFO] [1543315468.031813338]: Bond broken, exiting
[ INFO] [1543315468.032129962]: Bond broken, exiting
[object_analytics/splitter-2] process has finished cleanly
log file: /home/user/.ros/log/5ecdc050-f231-11e8-adc1-1c1b0d23f337/object_analytics-splitter-2*.log
[object_analytics/segmenter-3] process has finished cleanly
log file: /home/user/.ros/log/5ecdc050-f231-11e8-adc1-1c1b0d23f337/object_analytics-segmenter-3*.log
[object_analytics/merger-4] process has finished cleanly
log file: /home/user/.ros/log/5ecdc050-f231-11e8-adc1-1c1b0d23f337/object_analytics-merger-4*.log
[object_analytics/detecter-6] process has finished cleanly
log file: /home/user/.ros/log/5ecdc050-f231-11e8-adc1-1c1b0d23f337/object_analytics-detecter-6*.log
all processes on machine have died, roslaunch will exit
shutting down processing monitor...
... shutting down processing monitor complete
done`

but running the demo_test of ros_opencl_caffe have no problem,and I checked if clCaffe is working properly,then i have got this:

`./tools/caffe device_query -gpu all
I1127 19:12:11.362037 23255 common.cpp:556] Total devices: 1
I1127 19:12:11.362148 23255 common.cpp:557] CUDA devices: 0
I1127 19:12:11.362166 23255 common.cpp:558] OpenCL devices: 1
I1127 19:12:11.362171 23255 common.cpp:582] Device id:                     0
I1127 19:12:11.362188 23255 common.cpp:584] Device backend:                OpenCL
I1127 19:12:11.362200 23255 common.cpp:586] Backend details:               NVIDIA Corporation: OpenCL 1.2 CUDA 9.0.368
I1127 19:12:11.362205 23255 common.cpp:588] Device vendor:                 NVIDIA Corporation
I1127 19:12:11.362298 23255 common.cpp:590] Name:                          GeForce GTX 1070
I1127 19:12:11.362434 23255 common.cpp:592] Total global memory:           8504868864
`
so i don't know where the issue is?

**Comment by yechun1** [comment]
is there any more details from the log file under /home/user/.ros/log?

**Comment by Peter9606** [comment]
@hyuegithub  NVidia GPU is not a proper device for clCaffe to work with, but Intel integrated GPU.

**Comment by hyuegithub** [comment]
The information of  `./home/user/.ros/log/5ecdc050-f231-11e8-adc1-1c1b0d23f337 like this:
[0m[ INFO] [1543315450.516730862]: Spinning until killed publishing /camera_link to /camera_rgb_frame[0m`
I am trying to reinstall clCaffe.

**Comment by hyuegithub** [comment]
Thanks for your reply,you means that I missed some information when installing clCaffe? or the version of clCaffe I installed is wrong? or is the problem with hardware devices?

At 2018-11-28 09:45:11, "Peter Han" <notifications@github.com> wrote:


@hyuegithub NVidia GPU is not a proper device for clCaffe to work with, but Intel integrated GPU.

**Comment by Peter9606** [comment]
@hyuegithub  by clCaffe, which version are you using exactly?

**Comment by yechun1** [comment]
@hyuegithub it seems the be loader issue, would you please remark the code of opencl in analytics_opencl_caffe.launch to and only launch OA modules, please attach all the logs print from console. 

```
<!--
      <include file="$(find opencl_caffe_launch)/launch/includes/nodelet.launch">
          <arg name="manager"                   value="$(arg manager)" />
          <arg name="name"                      value="detecter" />
          <arg name="input_topic"               value="$(arg input_rgb)" />
          <arg name="output_topic"              value="$(arg input_detection)" />
      </include>
-->
```

**Comment by hyuegithub** [comment]
I am so happy to find out the wrong place,The issue is that I missed the  one "intel-opencl-cpu-r4.1-61547.x86_64.rpm"when I installing the opencl SDK. Now it can recognizes two devices properly.

and I still not successfuly roslaunch the "roslaunch object_analytics_launch analytics_opencl_caffe.launch",occurred same error ` process has died ` .

so I continue to check clCaffe if properly work or not,I found when I running the  command
 `build/tools/caffe-fp16.bin  test -model models/yolo/yolo416/yolo_fused_test.prototxt  -phase TEST -iterations 1000000 -weights models/yolo/yolo416/fused_yolo.caffemodel -gpu 0`,
I got this error:
`terminate called after throwing an instance of 'viennacl::ocl::invalid_program_executable'
  what():  ViennaCL: FATAL ERROR: CL_INVALID_PROGRAM_EXECUTABLE.`

and I reinstall the viennacl ,but it does't solve this problem.so what should I do?

**Comment by Peter9606** [comment]
@hyuegithub  what you show above is as design. you cannot expect any more progress w/o 2D detection node running.

**Comment by Jiang1206** [comment]
> 我很高兴找到错误的地方，问题是我在安装opencl SDK时错过了“intel-opencl-cpu-r4.1-61547.x86_64.rpm”。现在它可以正确识别两个设备。
> 
> 我仍然没有成功地发布了roslaunch“roslaunch object_analytics_launch analytics_opencl_caffe.launch”，发生了同样的错误`process has died`。
> 
> 所以我继续检查clCaffe是否正常工作，我发现当我运行命令时
> `build/tools/caffe-fp16.bin test -model models/yolo/yolo416/yolo_fused_test.prototxt -phase TEST -iterations 1000000 -weights models/yolo/yolo416/fused_yolo.caffemodel -gpu 0`，
> 我收到此错误：
> `terminate called after throwing an instance of 'viennacl::ocl::invalid_program_executable' what(): ViennaCL: FATAL ERROR: CL_INVALID_PROGRAM_EXECUTABLE.`
> 
> 我重新安装了viennacl，但它没有解决这个问题。那我该怎么办？

I also encountered the same problem. Has this problem been solved?

**Comment by hyuegithub** [comment]
> 
> 
> > 我很高兴找到错误的地方，问题是我在安装opencl SDK时错过了“intel-opencl-cpu-r4.1-61547.x86_64.rpm”。现在它可以正确识别两个设备。
> > 我仍然没有成功地发布了roslaunch“roslaunch object_analytics_launch analytics_opencl_caffe.launch”，发生了同样的错误`process has died`。
> > 所以我继续检查clCaffe是否正常工作，我发现当我运行命令时
> > `build/tools/caffe-fp16.bin test -model models/yolo/yolo416/yolo_fused_test.prototxt -phase TEST -iterations 1000000 -weights models/yolo/yolo416/fused_yolo.caffemodel -gpu 0`，
> > 我收到此错误：
> > `terminate called after throwing an instance of 'viennacl::ocl::invalid_program_executable' what(): ViennaCL: FATAL ERROR: CL_INVALID_PROGRAM_EXECUTABLE.`
> > 我重新安装了viennacl，但它没有解决这个问题。那我该怎么办？
> 
> I also encountered the same problem. Has this problem been solved?

no, I've give up!I haven't install CUDA, so I can't run damo in GPU. Inaddition, I think this README not enough detial in the aspect of environmental configuration. Usually，I encountered fail to run successfully  launch file, the most common reason is that there is a conflict between nodes or a node fails to run successfully (variable input error), the program itself is wrong (the wrong assignment of variables ).

**Comment by Peter9606** [comment]
@hyuegithub  To run demo in GPU, you don't have to install CUDA as clCaffe running on Intel integrated GPU.
This project is flexible to have different 2D detection back-end, even your own solution can be easily integrated into this project. I'd like to have some try on integrating other popular 2D backend. So could you tell me what's your favorite 2D detection solution?

**Comment by hyuegithub** [comment]
> @hyuegithub To run demo in GPU, you don't have to install CUDA as clCaffe running on Intel integrated GPU.
> This project is flexible to have different 2D detection back-end, even your own solution can be easily integrated into this project. I'd like to have some try on integrating other popular 2D backend. So could you tell me what's your favorite 2D detection solution?

mmdetection is seems pretty good for 2D detection.
